File: src/main/kotlin/utils/ScreenshotUtil.java
package utils;

import com.microsoft.playwright.Page;
import java.nio.file.Path;

public final class ScreenshotUtil {
    private ScreenshotUtil() {}

    public static Path take(Page page, String name) {
        Path p = Path.of("build", "screenshots", name + ".png");
        page.screenshot(new Page.ScreenshotOptions()
                .setPath(p)
                .setFullPage(true));
        return p;
    }
}

File: src/main/kotlin/utils/ScreenshotUtil.kt
package utils

import com.microsoft.playwright.Page
import java.nio.file.Path

object ScreenshotUtil {
    fun take(page: Page, name: String): Path {
        val p = Path.of("build", "screenshots", "$name.png")
        page.screenshot(Page.ScreenshotOptions()
            .setPath(p)
            .setFullPage(true))
        return p
    }
}

